kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/VirtualKafkaClusterReconciler.java

```diff
@@ -576,4 +572,16 @@ private VirtualKafkaCluster checkTlsConfigConsistency(Context<VirtualKafkaCluste
         return null;
     }
 
+    private static ClusterIP.Protocol getIngressProtocol(KafkaProxyIngress proxyIngress) {
+        ClusterIP clusterIP = proxyIngress.getSpec().getClusterIP();
+        if (clusterIP != null) {
+            return clusterIP.getProtocol();
+        }
+        LoadBalancer loadBalancer = proxyIngress.getSpec().getLoadBalancer();
+        if (loadBalancer != null) {
+            return ClusterIP.Protocol.TLS;
+        }
+        throw new IllegalStateException("No protocol could be determined for " + proxyIngress);
```

Comment:
Should we just assume TCP in this case? 

I guess its really just a theoretical exception as we shouldn't see a clusterIp resource without a protocol due to API server validation right?

Reply:
The CRD enforces that that is either clusterIp or loadBalancer, so I think the logic Rob has is right.